pkgs/development/web/nodejs/nodejs.nix

```diff
@@ -160,9 +160,14 @@ let
       ${if stdenv.buildPlatform.isGnu then ''
         ar -cqs $libv8/lib/libv8.a @files
       '' else ''
-        cat files | while read -r file; do
-          ar -cqS $libv8/lib/libv8.a $file
-        done
+        # llvm-ar supports response files, so take advantage of it if it’s available.
+        if [ "$(basename $(readlink -f $(command -v ar)))" = "llvm-ar" ]; then
```

Comment:
Looks fine to me too. I wonder though if there some way to find out whether `llvm-ar` is expected at eval time in nix so the block becomes:
```
${if stdenv.buildPlatform.isGnu || stdenv.buildPlatform.isLlvmAr} then ''
    ar -cqs $libv8/lib/libv8.a @files
'' else ''
    ...
''
```

Reply:
Not currently. We discussed in the Matrix room.

Reply:
Ideally we would set `isLLVM`, but Darwin isn’t a pure-LLVM platform. Maybe someday, but that’s still a ways off.